db/db_impl/db_impl_write.cc

```diff
@@ -1124,13 +1124,29 @@ IOStatus DBImpl::WriteToWAL(const WriteThread::WriteGroup& write_group,
     //    writer thread, so no one will push to logs_,
     //  - as long as other threads don't modify it, it's safe to read
     //    from std::deque from multiple threads concurrently.
+    //
+    // Sync operation should work with locked log_write_mutex_, because:
+    //   when DBOptions.manual_wal_flush_ is set,
+    //   FlushWAL function will be invoked by another thread.
```

Comment:
Thanks for review.  Already fix it.

Reply:
Thanks for the fix, will merge early this week. In the meantime, it would be great if you can rebase to reduce the chance of conflicts with others' code changes. Once we get an approval on Phabricator, we can merge it.